insights/parsers/journalctl.py

```diff
@@ -0,0 +1,154 @@
+        May 18 15:13:36 lxc-rhel68-sat56 wrapper[11375]: --> Wrapper Started as Daemon
+        May 18 15:13:36 lxc-rhel68-sat56 wrapper[11375]: Launching a JVM...
+        May 18 15:24:28 lxc-rhel68-sat56 yum[11597]: Installed: lynx-2.8.6-27.el6.x86_64
+        May 18 15:36:19 lxc-rhel68-sat56 yum[11954]: Updated: sos-3.2-40.el6.noarch
+
+    .. note::
+        Because journal timestamps by default have no year,
+        the year of the logs will be inferred from the year in your timestamp.
+        This will also work around December/January crossovers.
+
+    Examples:
+        >>> type(JournalAll)
+        <class 'insights.parsers.journalctl.JournalAll'>
+        >>> len(JournalAll.lines)
+        10
+        >>> bona_list = JournalAll.get('(root) LIST (root)')
+        >>> bona_list[0].get('message')
+        '(root) LIST (root)'
+    """
+    pass
+
+
+@parser(Specs.journalctl_header)
+class JournalctlHeader(LegacyItemAccess, CommandParser):
+    """
+    This command shows internal header information of the journal fields accessed.
+
+    Sample Output::
+
+        File Path: /run/log/journal/6bdaf92aa0754b53acbb1dbff7127e2b/system.journal
+        File ID: b1390ea69aa747e9ac5c597835c3c562
+        Machine ID: 6bdaf92aa0754b53acbb1dbff7127e2b
+        Boot ID: 082ada53f8184c4896c73101ad793eb5
+        Sequential Number ID: 435b0e30f47a46d8a2a2f9a42eae0aaf
+        State: ONLINE
+        Compatible Flags:
+        Incompatible Flags: COMPRESSED-LZ4
+        Header size: 240
+        Arena size: 8388368
+        Data Hash Table Size: 19904
+        Field Hash Table Size: 333
+        Rotate Suggested: no
+        Head Sequential Number: 74388 (12294)
+        Tail Sequential Number: 81647 (13eef)
+        Head Realtime Timestamp: Mon 2022-08-15 12:01:10 CST (5e63fae9e6e58)
+        Tail Realtime Timestamp: Wed 2022-08-17 18:38:48 CST (5e66d7852bb3e)
+        Tail Monotonic Timestamp: 1month 2w 3d 14h 19min 3.733s (3c647d7ce0f)
+        Objects: 19073
+        Entry Objects: 7260
+        Data Objects: 9297
+        Data Hash Table Fill: 46.7%
+        Field Objects: 52
+        Field Hash Table Fill: 15.6%
+        Tag Objects: 0
+        Entry Array Objects: 2462
+        Disk usage: 8.0M
+
+    Examples:
+        >>> type(journalctl_header)
+        <class 'insights.parsers.journalctl.JournalctlHeader'>
+        >>> journalctl_header.data[0]['File Path']
+        '/run/log/journal/6bdaf92aa0754b53acbb1dbff7127e2b/system.journal'
+        >>> journalctl_header.data[0]['File ID']
+        'b1390ea69aa747e9ac5c597835c3c562'
+
+    Attributes:
+        data (list): List of Dictionary, which containing each of the key:value pairs from the command output.
+
+    Raises:
+        SkipException: raised if data is invalid.
+    """
+
+    def parse_content(self, content):
+        body = {}
+        self.data = []
+
+        for line in get_active_lines(content):
+            if ':' in line:
```

Comment:
@TZ3070 Have you tried [split_kv_pairs](https://insights-core.readthedocs.io/en/latest/api_index.html?highlight=split_kv_pair#insights.parsers.split_kv_pairs)?

Reply:
I have updated to inherit from Syslog and not parsed the content